Hazel/src/Platform/OpenGL/OpenGLShader.cpp

```diff
@@ -97,7 +97,7 @@ namespace Hazel {
 			size_t eol = source.find_first_of("\r\n", pos); //End of shader type declaration line
 			HZ_CORE_ASSERT(eol != std::string::npos, "Syntax error");
 			size_t begin = pos + typeTokenLength + 1; //Start of shader type name (after "#type " keyword)
-			std::string type = source.substr(begin, eol - begin);
+			std::string_view type = source.substr(begin, eol - begin);
```

Comment:
Could be wrong, but isn't this "viewing" a deallocated substring?

Reply:
The result of `std::string_view::substr` is a `std::string_view`, so no deallocated string/temporary usage here

Reply:
> The result of `std::string_view::substr` is a `std::string_view`, so no deallocated string/temporary usage here

right, it's a substring of a `string_view`... missed that one